### DISCUSSION TITLE
Laser not lasering

### DISCUSSION THREAD

**Discussion by felanart** [arkypita/LaserGRBL#1700] [Q&A]
I know, this one again. But, I've spent the last week going through all of the answers posted here, and none of them have solved my issue. So, here we are.
As far as I can tell, I have laserGRBL properly installed and configured. As near as I can determine, I have the generic laser machine set up the right way. But, apparently, I have missed something, somewhere. 
I bought the laser system used from a friend. She said that the few times she used it, it worked fine. But, then it sat in storage for more than a year. I set it up using the software downloaded from the website she gave me. The laser did not come on. I uninstalled and reinstalled the software. Still no laser output. I doublechecked the wiring, based on the manufacturer's sketchy photographs. Still no laser. So, I decided to assume that the laser module was probably bad. I ordered a 40W from Lasertree on eBay. It is installed. I downloaded and installed a new version of laserGRBL (from the laserGRBL website). Nothing. Well, not entirely nothing:

The focus and frame work. But, when I hit the run button, no laser. The mechanism moves left and right, round and round, tracing where the laser should be burning. But, no laser. So, what am I doing wrong? Please help!

 I am going to attach photos which I hope will make it easier to rescue me.

![20220103_134112](https://user-images.githubusercontent.com/97055920/147970554-9d98db5f-e5b5-4445-a9c0-5e11cf9df3ba.jpg)
![20220103_134121](https://user-images.githubusercontent.com/97055920/147970558-c30a2bf5-da03-495e-875a-749e69fdb60d.jpg)
![20220103_134123](https://user-images.githubusercontent.com/97055920/147970565-fa2ca51b-d474-466d-9322-91416b0a83ed.jpg)
![20220103_134214](https://user-images.githubusercontent.com/97055920/147970571-b9c14b17-7089-4cb6-b4ea-f91a26541dae.jpg)
![20220103_134217](https://user-images.githubusercontent.com/97055920/147970574-8d67c998-46af-4cb0-be4a-cef48f2b1f82.jpg)
![20220103_134220](https://user-images.githubusercontent.com/97055920/147970577-ddf59d41-f615-4fd4-b3ea-4c3d2012c168.jpg)
![20220103_134225](https://user-images.githubusercontent.com/97055920/147970581-f08d2549-5cf8-45e2-a1b6-726609e745f9.jpg)
![20220103_135631](https://user-images.githubusercontent.com/97055920/147970586-4f6c546d-2987-4c60-a5a6-405f4f82f480.jpg)
![20220103_135654](https://user-images.githubusercontent.com/97055920/147970590-d554be16-50fd-4a3c-a3a3-d96581af4c2c.jpg)
![20220103_135707](https://user-images.githubusercontent.com/97055920/147970595-5e0261e7-1747-4af9-9854-8cd55d8ecee9.jpg)
![screen1](https://user-images.githubusercontent.com/97055920/147970600-7b047320-dab5-44c1-a35c-282b1324349e.jpg)
![screen2](https://user-images.githubusercontent.com/97055920/147970601-4f6b3ca4-47f3-49df-a1b0-92a40c5f7bf0.jpg)

Thank you for getting this far, and for considering helping me.

Bob

**Comment by Tribo-2dot0** [comment]
In GRBL settings try changing $33 to 5000 and see if anything changes.

**Reply by felanart** [reply]
Thank you. This had zero effect. :-(

**Reply by felanart** [reply]
And on going back in to the configuration, it changed itself to $33 = 136    ???

**Reply by arkypita** [reply]
Don't know what $33 is because it's not part of standard grbl $ codes. https://github.com/gnea/grbl/wiki/Grbl-v1.1-Configuration#grbls-xval-settings-and-what-they-mean

I suggest to restore the original value.

**Reply by Tribo-2dot0** [reply]
From what I read, this $33 is to set the PWM frequency, where the default value should be 5000 for engraving and cutting thin materials and a value of 2500 for more consistent cuts, so I suggested, but this is all part of such a from Grbl-LPC, I'm not sure what it is...
https://github.com/gnea/grbl-LPC
my Grbl code is standard (without $33).
I know that the PWM voltage was verified to be 2.5vdc and that the problem is there, but I would also flash Grbl through the Tools > Flash Grbl Firmware menu... since it will work with LaserGrbl, leave everything standardized .

**Comment by StuartB4** [comment]
You may need to swap the wires around on the laser connector at the main control board socket.
The main cnc board you have does'nt say which pin is Ground, PWM or 12+ Power. It just says Laser.
You can check which pin is ground with a multimeter, it should be connected to the main ground of the board and even the metal shield on the top of the usb connector. 

Unless the laser is wired the same way as the one you replaced, then the problem is somewhere else.

**Reply by felanart** [reply]
The center pin on the GRBL control board is ground. The center pin on the laser module board is ground. (both checked by multimeter) When I turn the connector around, neither the fan nor the focus light work, so it appears to be connected the right way. :-(

**Comment by arkypita** [comment]
TTL/PWM pin (yellow cable) is the one that control your laser on/off: 0V=OFF, 5V=ON
You have a multimeter so you can check it.

Turn on with this command: M3 S1000 G1 F1
Turn off with this command: M5

You can send this command via manual gcode box, under the progress bar. Please wear safety goggles

**Reply by arkypita** [reply]
If you measure 5V between GND and PWM your laser should turn on at maximum power, if not... it's broken.
If you don't measure 5V the issue is on control board side.

**Reply by felanart** [reply]
It is measuring at 2.5 vdc on the control board.

**Reply by arkypita** [reply]
Can you measure it with laser modules detached?

**Reply by felanart** [reply]
I did. I detached the wire from the control board, and measured it on the pins.

**Reply by arkypita** [reply]
2.5v is a little low value to be able to turn on driver logic. I think you have found your problem

**Reply by felanart** [reply]
Well, the major symptom anyway. :-) Now, what is the cause and the solution? (no, that was not a request for an answer, that was a lamentation of future difficulties)  Watch this space, I will see what I can find and provide updates. :-)

Thank you for your help!

**Reply by arkypita** [reply]
I am an electronics expert, but I am not a fortune teller :-)
I hypothesize two possibilities: hardware failure or firmware issue.

Just to check if it's a firmware issue you can try to re-flash firmware on your board with grbl firmware.
Take note of your $ values (can be necessary to restore them later) and use LaserGRBL to flash your board. Use v1.1h, original 20190825

https://lasergrbl.com/usage/flash/

**Reply by arkypita** [reply]
Flash is available only when LaserGRBL is disconnected.

**Reply by felanart** [reply]
Found it...thanks!

**Reply by arkypita** [reply]
news?

**Reply by felanart** [reply]
Flashing the firmware had no effect.
After much frustration in trying to find a schematic (and remembering that all of my electronics tools had been given away years ago), I decided to order a new controller card. It should be here early next week.

**Reply by arkypita** [reply]
Load an high res picture of the board, taken from the front, so it's possible to see all component.

**Reply by felanart** [reply]
![20220106_071015](https://user-images.githubusercontent.com/97055920/148381775-502ad90d-1806-4489-9e6b-edefe3e9431b.jpg)
![20220106_071017](https://user-images.githubusercontent.com/97055920/148381798-4ed2a98a-9d72-46e5-9076-232a17cb2c2c.jpg)
![20220106_071025](https://user-images.githubusercontent.com/97055920/148381824-ec8f4982-db9c-4b9d-bc5c-96aec4bbfa15.jpg)
![20220106_071145](https://user-images.githubusercontent.com/97055920/148381838-42322a78-fe5d-43d4-bcc7-fb456e35b3a9.jpg)
![20220106_071146](https://user-images.githubusercontent.com/97055920/148381850-2be6d874-ae7c-406f-a7cd-6e1654288149.jpg)

**Comment by arkypita** [comment]
I see no voltage converter on this board, so I can assume that the Atmega328 chip is directly powered by USB port.
You can verify it by check continuity between USB ground and Atmega328, and between USB 5V and Atmega328 VCC (just check one of them). Measuring voltage between Atmega328 GND and VCC should give you 5V, same as if measured on your board headers where you read VCC/GND.

![image](https://user-images.githubusercontent.com/8782035/148433685-ce369178-e9de-4882-a945-980a676d48e5.png)

As per GRBL documentation, PWM pin is D11 that in arduino nano is connected to Atmega328 PB3

![image](https://user-images.githubusercontent.com/8782035/148433330-dd80a72d-7c11-4f32-aa57-41815d3e7975.png)
![image](https://user-images.githubusercontent.com/8782035/148433410-b374a5a7-8bdb-470f-b985-f47f9bbd261c.png)
![image](https://user-images.githubusercontent.com/8782035/148433130-89bc3e91-67f5-45c8-9601-fe0e6e83471f.png)

In my opinion should be a direct connection from PB3 to your PWM pin.
Check it with multimeter (continuity and voltage)

**Reply by felanart** [reply]
I got the new control board. I am assuming I need to "flash the firmware", but, I'm not clear on which option to choose. The board is branded "Doesbot", but the flashing options are "Arduino Uno" and "Arduino Nano". So, which one do I choose? :-)

(This whole process was supposed to be a simple plug and play, not an adventure novel.)
![boardname](https://user-images.githubusercontent.com/97055920/149187568-b91ac3e3-b05e-43f6-b6c2-718ed99563e1.jpg)
![flash1](https://user-images.githubusercontent.com/97055920/149187596-36adf206-5ded-4eeb-aebb-ccc880522d4a.jpg)
![flash2](https://user-images.githubusercontent.com/97055920/149187603-1def3cdd-62a4-42ae-97be-205f21009652.jpg)

**Reply by felanart** [reply]
![newboard1](https://user-images.githubusercontent.com/97055920/149187813-1618dbb4-7fe7-4807-abe0-60d4c0b44f4e.jpg)

**Comment by StuartB4** [comment]
I would use Nano. Usually thats the option for controllers with built in Arduino chip.
Then select v1.1 original 8/25/2019 firmware, unless you have homing switches then select the one with homing.

**Reply by arkypita** [reply]
Sure that this board has no grbl firmware pre-installed?

**Reply by felanart** [reply]
No, I am not sure.

**Reply by Tribo-2dot0** [reply]
looking at the manufacturer's website, it seems to me that it already comes with grbl 1.1 installed

https://doesbot.com/products/cnc-accessories/controller-and-driver/GRBL-1-1-Control-board-CNC-Controller-Router-CNC-Engraving-Machine-Control-Board-3018-3-Axis-Control-Laser-Board-Corexy-v3-0.html

**Comment by felanart** [comment]
Final update. New GRBL control board had issues. It turns out that the PWM/GND/VDC plug on the board is pinned opposite of the one on the laser control board. A little rewiring of the connector cable solved that. Everything on the hardware side is now working famously. I have begun experimenting with the software so I can make it create what I want created. That's going to take some time, but it is working.

Thank you for your help!
